slackbridge/bots.py

```diff
@@ -129,17 +131,48 @@ def log(self, method, message):
         return method(full_message)
 
     def signedOn(self):
-        # If already registered, auth in
-        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
-        # And if not, register for the first time
-        self.msg('NickServ', 'GROUP {} {}'.format(self.target_group_nick,
-                                                  self.nickserv_password))
+        self.nickserv_auth()
+
         for channel_name in self.joined_channels:
             self.log(log.msg, 'Joining #{}'.format(channel_name))
             self.join(channel_name)
 
         self.away('Default away for startup.')
 
+    def nickserv_auth(self):
+        # If already registered, authenticate yourself to Nickserv
+        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
+        # And if not, register for the first time,
+        # if the nickname is the intended one.
+        if self.nickname == self.intended_nickname:
+            self.msg('NickServ', 'GROUP {} {}'.format(
+                self.target_group_nick, self.nickserv_password))
+
+    # Called whenever the client wants to set a nickname, such as on startup
+    # or if there's a collision.
+    def setNick(self, nickname):
+        super().setNick(nickname)
+        self.nickserv_auth()
+
+    def nickChanged(self, nick):
+        super().nickChanged(nick)
+        if nick != self.intended_nickname:
+            self.log(log.msg,
+                     'Attempting to change nick to {} in 10 seconds.'.format(
+                         self.intended_nickname))
```

Comment:
I think formatting this like:

```
self.log(
    log.msg,
    'Attempting to change nick to {} in 10 seconds.'.format(self.intended_nickname),
)
```

or like this if the line is still too long:

```
self.log(
    log.msg,
    'Attempting to change nick to {} in 10 seconds.'.format(
        self.intended_nickname,
    ),
)
```
would make this a little cleaner.